### PR TITLE
agent: use macro to simplify parse_cmdline function in config.rs

### PR DESCRIPTION
**Pull request by liubin**
In function parse_cmdline there are some similar codes, if we want
to add more commandline arguments, the code will grow too long.
Use macro can reduce some codes with the same logic/processing.

Fixes: #914

Signed-off-by: bin liu <bin@hyper.sh>

### CODE REVIEW

File: src/agent/src/config.rs [hunk 1]
```diff
@@ -185,32 +213,32 @@ fn get_hotplug_timeout(param: &str) -> Result<time::Duration> {
...
+        return Ok(false);
     }
 
     let v = fields[1];
 
     // bool
     let t: std::result::Result<bool, std::str::ParseBoolError> = v.parse();
     if t.is_ok() {
-        return t.unwrap();
+        return Ok(t.unwrap());
```
Comment (bergwolf) [inline src/agent/src/config.rs:228]: same here.

File: src/agent/src/config.rs [hunk 2]
```diff
@@ -185,32 +213,32 @@ fn get_hotplug_timeout(param: &str) -> Result<time::Duration> {
...
+        return Ok(false);
     }
 
     // only `0` returns false, otherwise returns true
-    match i.unwrap() {
+    Ok(match i.unwrap() {
         0 => false,
         _ => true,
-    }
+    })
```
Comment (bergwolf) [inline src/agent/src/config.rs:241]: This `is_err()` `unwrap()` pattern can be replaced by a `if let` statement that is cleaner and less code.

**Review by bergwolf — APPROVED** [review]
lgtm!

**Review by lifupan — APPROVED** [review]

**Pull request merged**